File: api-backend/nodejs/retrieve_billing_and_payment_information.js
// Config
const API_HOST = 'https://yourportal.onlyoffice.com';
const API_KEY = 'your_api_key';

const HEADERS = { Authorization: API_KEY };

// Step 1: Customer info
async function getCustomerInfo() {
  const res = await fetch(`${API_HOST}/api/2.0/portal/payment/customerinfo`, { headers: HEADERS });
  if (!res.ok) {
    const t = await res.text();
    console.log(`Failed to get customer info: ${res.status} - ${t}`);
    return null;
  }
  const data = await res.json();
  const info = data?.response ?? {};
  console.log('Customer Info:', info);
  return info;
}

// Step 2: Customer balance
async function getCustomerBalance() {
  const res = await fetch(`${API_HOST}/api/2.0/portal/payment/customer/balance`, { headers: HEADERS });
  if (!res.ok) {
    const t = await res.text();
    console.log(`Failed to get balance: ${res.status} - ${t}`);
    return null;
  }
  const data = await res.json();
  const balance = data?.response ?? {};
  console.log('Balance:');
  for (const sub of balance.subAccounts ?? []) {
    console.log(`- ${sub.currency}: ${sub.amount}`);
  }
  return balance;
}

// Step 3: Current quota
async function getCurrentQuota() {
  const res = await fetch(`${API_HOST}/api/2.0/portal/payment/quota`, { headers: HEADERS });
  if (!res.ok) {
    const t = await res.text();
    console.log(`Failed to get current quota: ${res.status} - ${t}`);
    return null;
  }
  const data = await res.json();
  const q = data?.response ?? {};
  console.log(`Current Quota: ${q.title}, Trial: ${q.trial}, Due: ${q.dueDate}`);
  return q;
}

// Step 4: Available quotas
async function getAvailableQuotas() {
  const res = await fetch(`${API_HOST}/api/2.0/portal/payment/quotas`, { headers: HEADERS });
  if (!res.ok) {
    const t = await res.text();
    console.log(`Failed to get available quotas: ${res.status} - ${t}`);
    return null;
  }
  const data = await res.json();
  const list = data?.response ?? [];
  console.log('Available Quotas:');
  for (const q of list) {
    const price = q.price ?? {};
    console.log(`- ${q.title} - ${price.value} ${price.isoCurrencySymbol}`);
  }
  return list;
}

// Step 5: Auto Top-Up settings
async function getTopupSettings() {
  const res = await fetch(`${API_HOST}/api/2.0/portal/payment/topupsettings`, { headers: HEADERS });
  if (!res.ok) {
    const t = await res.text();
    console.log(`Failed to get topup settings: ${res.status} - ${t}`);
    return null;
  }
  const data = await res.json();
  const settings = data?.settings ?? data?.response ?? {};
  console.log('Auto Top-Up Settings:');
  console.log(`- Enabled: ${settings.enabled}`);
  console.log(`- Min Balance: ${settings.minBalance}`);
  console.log(`- Up To: ${settings.upToBalance}`);
  console.log(`- Currency: ${settings.currency}`);
  console.log(`- Last Modified: ${settings.lastModified}`);
  return settings;
}

// Step 6: Supported currencies
async function getSupportedCurrencies() {
  const res = await fetch(`${API_HOST}/api/2.0/portal/payment/currencies`, { headers: HEADERS });
  if (!res.ok) {
    const t = await res.text();
    console.log(`Failed to get currencies: ${res.status} - ${t}`);
    return null;
  }
  const data = await res.json();
  const list = data?.response ?? [];
  console.log('Supported Currencies:');
  for (const c of list) {
    console.log(`- ${c.isoCurrencySymbol} - ${c.currencyNativeName}`);
  }
  return list;
}

// Run
(async () => {
  await getCustomerInfo();
  await getCustomerBalance();
  await getCurrentQuota();
  await getAvailableQuotas();
  await getTopupSettings();
  await getSupportedCurrencies();
})();

File: api-backend/python/samples/retrieve_billing_and_payment_information.py
'''
Example: Retrieve billing and payment information

This example demonstrates how to read billing-related information in ONLYOFFICE DocSpace via the API:
customer profile, balance by currency, current quota, available quotas, auto top-up settings, and supported currencies.

Using methods:
- GET /api/2.0/portal/payment/customerinfo (https://api.onlyoffice.com/docspace/api-backend/usage-api/get-customer-info/)
- GET /api/2.0/portal/payment/customer/balance (https://api.onlyoffice.com/docspace/api-backend/usage-api/get-customer-balance/)
- GET /api/2.0/portal/payment/quota (https://api.onlyoffice.com/docspace/api-backend/usage-api/get-quota-payment-information/)
- GET /api/2.0/portal/payment/quotas (https://api.onlyoffice.com/docspace/api-backend/usage-api/get-payment-quotas/)
- GET /api/2.0/portal/payment/topupsettings (https://api.onlyoffice.com/docspace/api-backend/usage-api/get-tenant-wallet-settings/)
- GET /api/2.0/portal/payment/currencies (https://api.onlyoffice.com/docspace/api-backend/usage-api/get-payment-currencies/)
'''

import requests

API_HOST = 'https://yourportal.onlyoffice.com'
API_KEY = 'your_api_key'

HEADERS = {'Authorization': API_KEY}

def get_customer_info():
  r = requests.get(f'{API_HOST}/api/2.0/portal/payment/customerinfo', headers=HEADERS)
  print("Customer Info:", r.json().get('response', {}))

def get_customer_balance():
  r = requests.get(f'{API_HOST}/api/2.0/portal/payment/customer/balance', headers=HEADERS)
  balance = r.json().get('response', {})
  print("Balance:")
  for sub in balance.get('subAccounts', []):
    print(f"- {sub.get('currency')}: {sub.get('amount')}")

def get_current_quota():
  r = requests.get(f'{API_HOST}/api/2.0/portal/payment/quota', headers=HEADERS)
  quota = r.json().get('response', {})
  print(f"Current Quota: {quota.get('title')}, Trial: {quota.get('trial')}, Due: {quota.get('dueDate')}")

def get_available_quotas():
  r = requests.get(f'{API_HOST}/api/2.0/portal/payment/quotas', headers=HEADERS)
  print("Available Quotas:")
  for q in r.json().get('response', []):
    price = q.get('price', {})
    print(f"- {q.get('title')} - {price.get('value')} {price.get('isoCurrencySymbol')}")

def get_topup_settings():
  r = requests.get(f'{API_HOST}/api/2.0/portal/payment/topupsettings', headers=HEADERS)
  settings = r.json().get('settings', {})
  print("Auto Top-Up Settings:")
  print(f"- Enabled: {settings.get('enabled')}")
  print(f"- Min Balance: {settings.get('minBalance')}")
  print(f"- Up To: {settings.get('upToBalance')}")
  print(f"- Currency: {settings.get('currency')}")
  print(f"- Last Modified: {settings.get('lastModified')}")

def get_supported_currencies():
  r = requests.get(f'{API_HOST}/api/2.0/portal/payment/currencies', headers=HEADERS)
  print("Supported Currencies:")
  for c in r.json().get('response', []):
    print(f"- {c.get('isoCurrencySymbol')} - {c.get('currencyNativeName')}")

if __name__ == '__main__':
  get_customer_info()
  get_customer_balance()
  get_current_quota()
  get_available_quotas()
  get_topup_settings()
  get_supported_currencies()
